### ISSUE TITLE
trie: cache encoding of nodes on disk

### ISSUE THREAD

**Issue by qdm12** [ChainSafe/gossamer#2322]
## Task summary

For now we cache the encoding of trie nodes in memory for every node.
We should instead write the cached encoding to disk and fetch it from disk when needed.

- Modify code to store node hash <-> encoding on encoding
- Modify code to delete node hash <-> encoding when node is modified or removed
- Modify code to fetch encoding using the node hash (kept in memory) when encoding
- Memory usage improvement measured and documented
- Speed performance degradation measured and documented

**Comment by noot** [comment]
@qdm12 same with this, do we still need this?

**Comment by qdm12** [comment]
This depends on #2229 which I'll finish up soon. If the number of total nodes in the tries in the program correlate with our increased memory usage, then we probably want to do both #2323 (store the `Value` byte slice of a node) and this issue (store the cached encoding and digest of a node).

**Comment by qdm12** [comment]
This is no longer needed, having the whole state trie (and deltas from forks before finalization) does not use much memory (~100MB at the tip). See #1973 for an update on memory usage. Let's close this for now, we can re-open later when we need it (i.e. for parachains with superfat tries)
